packages/shared/lib/sdk/sync.ts

```diff
@@ -912,10 +925,49 @@ export class NangoAction {
         }
 
         if (response.status > 299) {
-            logger.error(`Request to persist API (log) failed: errorCode=${response.status} response='${JSON.stringify(response.data)}'`, this.stringify());
+            logger.error(
+                `Request to persist API (log) failed: errorCode=${response.status} response='${JSON.stringify(response.data)}' log=${JSON.stringify(log)}`,
+                this.stringify()
+            );
             throw new Error(`Failed to log: ${JSON.stringify(response.data)}`);
         }
     }
+
+    private logAPICall(res: AxiosResponse): AxiosResponse {
+        if (!res.config.url) {
+            return res;
+        }
+
+        const valuesToFilter: string[] = [
```

Comment:
Just a small thought for the future: Could we compute our filter array once up front before any runner work and have the logger lean on it across all logging?

Reply:
agreed. The valuesToFilter should not changed within a single execution

Reply:
actually it's changing because connection's credentials can change during execution, but the value could be cached indeed. I'll add a comment